app/models/product_user_creator.rb

```diff
@@ -0,0 +1,21 @@
+class ProductUserCreator
+  def self.create(user:, product:, approver:)
+    product_user = ProductUser.new(
+      product: product,
+      user: user,
+      approved_by: approver.id,
+      approved_at: Time.zone.now,
+    )
+
+    product_user.transaction do
+      product_user.save && manage_training_request(product_user)
```

Comment:
Doesn't look like `#destroy!` exists in Rails 3.x. Guess I'm going with `training_request.destroy || raise ActiveRecord::Rollback`.